sqlglot/parser.py

```diff
@@ -5692,11 +5677,29 @@ def _parse_set(self, unset: bool = False, tag: bool = False) -> exp.Set | exp.Co
 
         return set_
 
-    def _parse_var_from_options(self, options: t.Collection[str]) -> t.Optional[exp.Var]:
-        for option in options:
-            if self._match_text_seq(*option.split(" ")):
-                return exp.var(option)
-        return None
+    def _parse_var_from_options(
+        self, options: t.Dict[str, t.Sequence[t.Sequence[str] | str]]
+    ) -> t.Optional[exp.Var]:
+        start = self._curr
+        if not start:
+            return None
+
+        option = start.text.upper()
+        continuations = options.get(option)
+
+        self._advance()
+        for keywords in continuations or []:
+            if isinstance(keywords, str):
+                keywords = (keywords,)
+
+            if self._match_text_seq(*keywords):
+                option = f"{option} {' '.join(keywords)}"
+                break
+        else:
+            if continuations:
+                self.raise_error(f"Unknown option {option}")
```

Comment:
is this more strict than it needs to be?

Reply:
yea

Reply:
Hmm maybe it's actually better to keep it, at least for now. I tried the retreat approach but e.g. for T-SQL we try to match a `)` after the option parser is called and that fails so the error message is cryptic - it says we're expecting a `)` but in reality we just failed to parse the option and retreated without consuming the tokens.

Reply:
k